toolchain/check/declaration_name_stack.h

```diff
@@ -78,7 +78,7 @@ class DeclarationNameStack {
     // The scope which qualified names are added to. For unqualified names in
     // an unnamed scope, this will be Invalid to indicate the current scope
     // should be used.
-    SemIR::NameScopeId target_scope_id = SemIR::NameScopeId::Invalid;
+    SemIR::NameScopeId target_scope_id;
```

Comment:
I think this works partly because NameContext is never default constructed right now, it'd all gone through Push(), but the defaults should probably be removed from other fields too.

Reply:
Instead changed `MakeEmptyNameContext` to make use of the default member initializers. If you'd prefer I remove them as suggested, I'm happy to; let me know.

Reply:
LGTM